src/Chapter1.hs

```diff
@@ -536,9 +537,14 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
-
+mid :: Int -> Int -> Int -> Int
+mid x y z 
+    | x < y && y < z && z > x = y
+    | x < y && y > z && z > x = z
+    | x > y && y < z && z > x = x
+    | x < y && y > z && z < x = x
+    | x < y && y < z && z < x = z
+    | x > y && y > z && z < x = y
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂

As you covered all cases by these checks, you actually know that if the input didn't match any previous cases, it means that the result is definitely `y`. You can skip checking, and write `otherwise = y` instead of this last line 👍🏼 
Also you can remove the first case(which returns `y`) as it also be covered by such `otherwise` case 🙂